code/bmpman/bmpman.cpp

```diff
@@ -567,6 +578,67 @@ void bm_free_data_fast(int n)
 	bmp->data = 0;
 }
 
+int bm_get_anim_frame(const int frame1_handle, float elapsed_time, const float divisor, const bool loop)
+{
+	int n = bm_get_cache_slot(frame1_handle, 1);
+	bitmap_entry *be = &bm_bitmaps[n];
+
+	if (be->info.ani.num_frames <= 1) {
+		// this is a still image
+		return 0;
+	}
+	int last_frame = be->info.ani.num_frames - 1;
+	bitmap_entry *last_framep = &bm_bitmaps[n + last_frame];
+
+	if (elapsed_time < 0.0f) {
+		elapsed_time = 0.0f;
+	}
+
+	int frame = 0;
+	// variable frame delay animations
+	if (be->info.ani.apng.is_apng == true) {
+		if (divisor > 0.0f) {
+			// scale to get the real elapsed time
+			elapsed_time = elapsed_time / divisor * last_framep->info.ani.apng.frame_delay;
+		}
+
+		if (loop == true) {
+			if (last_framep->info.ani.apng.frame_delay != 0.0f) {    // prevent infinite loops, which can occur if this is called before the anim is locked
+				while (elapsed_time >= last_framep->info.ani.apng.frame_delay) {
```

Comment:
I think this code could be replaced with `elapsed_time = fmod(elapsed_time, last_framep->info.ani.apng.frame_delay)`.